### ISSUE TITLE
Add validation for new undefined native symbols in hosting binaries

### ISSUE THREAD

**Issue by AaronRobinsonMSFT** [dotnet/core-setup#5368]
A check should be made to our native binaries when a build produces new undefined symbols. This would have been an early warning and possibly prevented https://github.com/dotnet/coreclr/issues/22653.

**Comment by omajid** [comment]
corefx uses https://github.com/dotnet/corefx/blob/master/src/Native/Unix/verify-so.sh to solve this issue on *nix.

**Comment by AaronRobinsonMSFT** [comment]
@omajid Yep. @janvorli suggested this as an option offline. One of the ways we can validate, but I also want to investigate some other options using compiler flags.

**Comment by janvorli** [comment]
@AaronRobinsonMSFT maybe this linker option would work:
```
--no-allow-shlib-undefined  Do not allow unresolved references in shared libs
```

**Comment by AaronRobinsonMSFT** [comment]
Yeah I was thinking of that one as well. My bigger concern here is the issue that failed was related to cross compilation. This makes the `ldd` approach not entirely applicable since it won't be on the actual OS that the image will be running. Instead I may need to use something like `objdump` to see what symbols are actually undefined and then diff them.

**Comment by janvorli** [comment]
Right, the ldd way would not work for cross compilation.

**Comment by AaronRobinsonMSFT** [comment]
I spent today trying to figure a reasonable way to validate these issues for cross-compiled binaries and I don't see a great way to do it. Using `ldd` obviously won't work, `objdump` would require having baseline files, and using one of the compiler flags would require fix all current undefined symbols. Even if all of these were done, the cross compile scenario still wouldn't be mitigated. This has led me to believe we need to add support for testing all supported platforms to core-setup.

Closing in deference to https://github.com/dotnet/core-setup/issues/5379
